CMakeLists.txt

```diff
@@ -532,10 +532,11 @@ if(WITH_REDUCED_MEM)
     message(STATUS "Configured for reduced memory environment")
 endif()
 
+set(ARCHDIR "arch/generic")
 
 set(ZLIB_ARCH_SRCS)
-set(ZLIB_ARCH_HDRS)
-set(ARCHDIR "arch/generic")
+set(ZLIB_ARCH_HDRS ${ARCHDIR}/generic_features.h)
```

Comment:
I think this should instead be a separate variable set to `arch/generic` (`GENERIC_ARCHDIR`?), since `ARCHDIR` is only set to `arch/generic` temporarily as a fallback. And the setting of `ARCHDIR` should be moved down to line 554, in the else() block.

Reply:
Done.